Sources/ArgumentParser/Parsable Types/CommandConfiguration.swift

```diff
@@ -62,6 +76,58 @@ public struct CommandConfiguration: Sendable {
   /// or `commandName` itself if provided.
   public var aliases: [String]
 
+  /// Creates the configuration for a command.
+  ///
+  /// - Parameters:
+  ///   - commandName: The name of the command to use on the command line. If
+  ///     `commandName` is `nil`, the command name is derived by converting
+  ///     the name of the command type to hyphen-separated lowercase words.
+  ///   - abstract: A one-line description of the command.
+  ///   - usage: A custom usage description for the command. When you provide
+  ///     a non-`nil` string, the argument parser uses `usage` instead of
+  ///     automatically generating a usage description. Passing an empty string
+  ///     hides the usage string altogether.
+  ///   - discussion: A longer description of the command.
+  ///   - version: The version number for this command. When you provide a
+  ///     non-empty string, the argument parser prints it if the user provides
+  ///     a `--version` flag.
+  ///   - shouldDisplay: A Boolean value indicating whether the command
+  ///     should be shown in the extended help display.
+  ///   - subcommands: An array of the types that define subcommands for the
+  ///     command.
+  ///   - defaultSubcommand: The default command type to run if no subcommand
+  ///     is given.
+  ///   - helpNames: The flag names to use for requesting help, when combined
+  ///     with a simulated Boolean property named `help`. If `helpNames` is
+  ///     `nil`, the names are inherited from the parent command, if any, or
+  ///     are `-h` and `--help`.
+  ///   - aliases: An array of aliases for the command's name. All of the aliases
+  ///     MUST not match the actual command name, whether that be the derived name
+  ///     if `commandName` is not provided, or `commandName` itself if provided.
```

Comment:
Something like this:
```suggestion
  ///     if `commandName` is not provided, or `commandName` itself if provided.
  ///   - groupedSubcommands: A builder closure that defines the subcommands
  ///     and their groups for this command.
```